souleb/source-controller
# refactor to take into account new reconciliation logic

- Bump libgit2 to 1.3.0 and git2go to V33. Downstream breaking changes introduced since git2go@V31: - git2go.ErrorCode was deprecated in favour of the native error type. - FetchOptions no longer expects a pointer, but rather the actual value of git2go.FetchOptions.
- Add libgit2 checkout test with ED25519 key This adds a test to detect any regression in libgit2's ED25519 key support. go-git supports ED25519 but not the current version of libgit2 used in flux. The updates to libgit2 in v1.2.0 adds support for ED25519. This test would help ensure the right version of libgit2 is used.
- Fix make test on arm64 runners The environment variables set at the Makefile were causing go install to yield a corrupted file for setup-envtest. To fix the issue, such operation is now always executed in a clean bash.
- Update libgit2 attributions
- Upgrade libgit2 to libgit2-1.3.0-2
- Add pjbgf to Maintainers
- Upgrade controller-runtime to v0.11.1 Fix for CVE-2022-21698 by upgrading the trasient dependency github.com/prometheus/client_golang.
- Upgrade docker/distribution to v2.8.0 Fixes https://github.com/advisories/GHSA-qq97-vm5h-rrhg
- git/libgit2: set CheckoutForce on branch strategy
- git/libgit2: assert proper test of default branch
- Introduce v1beta2 API package
- Switch to v1beta2 API package
- Introduce more explicit Condition types
- Implement new runtime interfaces, prepare testenv
- Introduce `artifactSet` to replace `hasArtifactUpdated`
- source: Add `GetRequeueAfter`
- Use new events and metrics helpers in main.go
- Move Artifact conditions to conditions
- Add gomega matcher for artifact
- api: Embed runtime.Object in Source interface
- Add internal packages error and reconcile
- internal/util: introduce temp dir/path helpers
- Rewrite `GitRepositoryReconciler` to new standards
- Fixes to gitrepo reconciler tests
- controllers: Add more tests for reconcileArtifact
- Replace %q in messages with '%s'
- source: `GetRequeueAfter` in place of `GetInterval`
- Tweak logged messages
- gitrepo: reconcileInclude test assertion fixes
- gitrepo: Add tests for old conditions update
- gitrepo: test reconcileArtifact condtns & symlink
- Consolidate condition types into `FetchFailed`
- Wrap err with context instead of logging twice
- chore: ensure Git server dir is removed after test
- gitrepo: Fix SourceVerifiedCondition condition type
- gitrepo: Ignore patch error not found on delete
- gitrepo: Add more reconciler design improvements
- gitrepo: Use internal/util for creating temp dir
- gitrepo: Fix reconcileInclude() includes
- bucket: Replace GetInterval() with GetRequeueAfter()
- Rewrite `BucketReconciler` to new standards
- Consolidate condition types into `FetchFailed`
- BucketReconciler: Add reconcileArtifact tests
- Add more reconcileMinioSource test cases
- Add bucket controller tests for reconcileGCPSource
- bucket: Ignore patch error not found on delete
- bucket: Add more reconciler design improvements
- helmrepo: Replace GetInterval() with GetRequeueAfter()
- Rewrite `HelmRepositoryReconciler` to new standards
- helmrepo: Add more reconciler design improvements
- helmchart: Replace GetInterval() with GetRequeueAfter()
- HelmChartReconciler refactor
- Rewrite HelmChartReconciler tests
- controllers: tweak events and logging
- controllers: remove legacy test suite
- api: remove obsolete constants
- Consolidate result conversion and computation
- internal/helm: ErrChartReference on local load err
- controllers: only handle BuildError
- controllers: HelmChart Reconcile test
- controllers: cleanup TestStorageCopyFromPath files
- controllers: truncate temporary cached Helm index
- controllers: replace `logr.NewContext` shims
- storage: Return details about the deleted items
- Make generic SummarizeAndPatch()
- Update API descriptions and messages to be consistent
- gitrepo: Use commit msg in NewArtifact message
- bucket: Make NewArtifact event more informative
- build: update meta API and pkg/runtime to non RC
- Use field owner in the patch helper
- Add kstatus client conformance tests
- fuzz: Update to use v1beta2 APIs
- helmrepo: Make NewArtifact event human friendly
- Update object API version in the sample configs
- api: Move Status in CRD printcolumn to the end
- Update github.com/sosedoff/gitkit (indirect) to v0.3.0 (CVE fix) Fixes https://snyk.io/vuln/SNYK-GOLANG-GITHUBCOMSATORIGOUUID-72488 in the indirect dependency github.com/satori/go.uuid.
- Remove redundant reconciling in reconcileArtifact
- Implement Size field on archived artifacts
- Introduce BucketProvider interface
- Update to patched version of Helm 3.8.0
- Reuse transport for helm chart download Reuses the same transport across different helm chart downloads, whilst resetting the tlsconfig to avoid cross-contamination.
- Use uppercase TLS in error messages
- Improve TransportPool documentation
- Remove dependency to helm v2
- Update dependencies - runc v1.1.0 (CVE fix https://github.com/advisories/GHSA-fgv8-vj5c-2ppq) - containerd v1.6.1 (CVE fix https://github.com/containerd/containerd/security/advisories/GHSA-crp2-qrr5-8pq7)
- Add support for using an OCI image as source
- Add an OCI Registry client
- fix: only try login when auth is configured; trim "oci" scheme
- Add support for docker mediaTypes
- feat: add short name for ocirepository type
- feat: make error message a little more concrete
- feat: allow OCIRepositry as source kind for HelmCharts
- feat: reconcile HelmCharts sourced from OCIRepositories
- Adding a new API type OCIRegistry
- refactor to take into account the new reconciliation logic
